neo/Consensus/ConsensusContext.cs

```diff
@@ -108,10 +109,10 @@ public void Deserialize(BinaryReader reader)
             ViewNumber = reader.ReadByte();
             TransactionHashes = reader.ReadSerializableArray<UInt256>();
             Transaction[] transactions = reader.ReadSerializableArray<Transaction>(Block.MaxTransactionsPerBlock);
-            PreparationPayloads = reader.ReadNullableArray<ConsensusPayload>(Blockchain.MaxValidators);
-            CommitPayloads = reader.ReadNullableArray<ConsensusPayload>(Blockchain.MaxValidators);
-            ChangeViewPayloads = reader.ReadNullableArray<ConsensusPayload>(Blockchain.MaxValidators);
-            LastChangeViewPayloads = reader.ReadNullableArray<ConsensusPayload>(Blockchain.MaxValidators);
+            PreparationPayloads = ReadConsensusPayload(reader, ViewNumber);
+            CommitPayloads = ReadConsensusPayload(reader, ViewNumber);
+            ChangeViewPayloads = ReadConsensusPayload(reader, ViewNumber);
+            LastChangeViewPayloads = ReadConsensusPayload(reader, ViewNumber);
```

Comment:
I think we can't read the payloads in current view only.

Reply:
Why we could want other views?

Reply:
I think that check Preparation also should check the view, but I changed to this model because i thought that was faster
https://github.com/neo-project/neo/commit/51af5be4256879956339882f81585e8ffc3c3317

Reply:
Preparations can't be in different views. Because:

https://github.com/neo-project/neo/blob/72be7f6a26f3ed222f1ad5dc1b4391c62fa4f67d/neo/Consensus/ConsensusContext.cs#L405

https://github.com/neo-project/neo/blob/72be7f6a26f3ed222f1ad5dc1b4391c62fa4f67d/neo/Consensus/ConsensusService.cs#L403-L406

https://github.com/neo-project/neo/blob/72be7f6a26f3ed222f1ad5dc1b4391c62fa4f67d/neo/Consensus/ConsensusService.cs#L474-L476

But we need `ChangeViewPayloads` from different views.

So the original is correct.

Reply:
Then i will revert the changes